packages/ifpack2/adapters/thyra/Thyra_Ifpack2PreconditionerFactory_def.hpp

```diff
@@ -76,6 +76,7 @@
 
 namespace Thyra {
 
+using Teuchos::rcp;
```

Comment:
This needs to be moved into the function bodies that use this.  See other comment on this.